hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -887,17 +876,19 @@ public boolean hasEnoughCapacity(int delta) {
      *
      * @return the item queue
      */
-    public Deque<QueueItem> getItemQueue() {
+    public PriorityQueue<QueueItem> getItemQueue() {
```

Comment:
Can you also add tests for all of the expected behaviour? Please take a look at the existing tests in `com.hazelcast.collection.impl.queue` and reuse all that apply to the priority queue as well.